### DISCUSSION TITLE
Support request: Extended SPI features for the Arduino library

### DISCUSSION THREAD

**Discussion by XanCraft21** [espressif/arduino-esp32#8174] [Ideas]
Hello, nobody probably cares about this, but i wanted to ask if you are able to add extended SPI functions and features to the Arduino SPI library. If you would be willing to add support for different SPI methods like Quad SPI or Octal SPI i will really appreciate that. The library currently does not support these and only supports normal SPI, and i don’t feel like decoding each bit in the registers to manually activate different SPI types. Thank you if you are able to add these.

**Comment by mrengineer7777** [comment]
You are correct. https://github.com/espressif/arduino-esp32/issues/7063

**Reply by XanCraft21** [reply]
Thank you. I did not see that issue open, sorry that i missed it. I hope that one day it gets added along with Octal SPI for the esp32 s3. I don’t think the esp32 s2 supports Octal SPI though.
